### PR TITLE
Implement Sentry SDK in Anomaly Detection

### PR DESCRIPTION
**Pull request by aayush-se**
- Enable Tracing for Store and Detect endpoints for Anomaly Detection
- Capture Exceptions through Sentry for each endpoint
- Traces are tagged with `organization_id` and `project_id` to filter in Sentry.io

Notes: 
- Explicit Metrics were not included as Performance monitoring captures relevant information (ie: duration of transactions and their spans)
- Logs for endpoints were already handled

Closes #881

### CODE REVIEW

**Comment by ram-senth** [comment]
@corps mind taking a look at this PR for adding Sentry logging and monitoring to anomaly detection end points?

**Comment by corps** [comment]
@ram-senth Yes, I'll take a look shortly.

File: src/seer/anomaly_detection/anomaly_detection.py [hunk 1]
```diff
@@ -88,11 +92,14 @@ def _update_anomalies(self, ts_external: List[TimeSeriesPoint], anomalies: TimeS
             )
 
     def detect_anomalies(self, request: DetectAnomaliesRequest) -> DetectAnomaliesResponse:
-        ts: List[TimeSeriesPoint] = (
-            self._online_detect(request.context)
-            if isinstance(request.context, AlertInSeer)
-            else self._batch_detect(request.context)
-        )
+        try:
```
Comment (corps) [inline src/seer/anomaly_detection/anomaly_detection.py:95]: Hmm, we shouldn't need this.  Actually, the `FlaskIntegration` already handles reporting unhandled exceptions for any request, we shouldn't need to do this except potentially in offline code paths (say, celery jobs).

**Review by corps — APPROVED** [review]
Nit about broad exception capturing.

**Pull request merged**